CONTRIBUTING.md

```diff
@@ -101,9 +101,87 @@ the Apache license found in the LICENSE file.
 A number of issues in the 
 [issue tracker](https://github.com/GoogleCloudPlatform/gcloud-intellij/issues/new)
 have been tagged as "[Help Wanted](https://github.com/GoogleCloudPlatform/gcloud-intellij/labels/help%20wanted)." 
-These are relatively small, self-contained changes that are good places to start. 
+These are relatively small, self-contained changes that are good places to start.
+
+## Project Utilities
+
+### `CloudToolsRule`
+We have a custom JUnit `TestRule` that reduces the amount of boilerplate code we have in our unit
+tests. Notably, this rule handles a lot of the set-up and tear-down required for a proper IntelliJ
+project context.
+
+For example:
+
+```
+@Rule public final CloudToolsRule cloudToolsRule = new CloudToolsRule(this);
+```
+
+A summary of its features can be found below.
+
+**Always:**
+1. Initializes the fields annotated with `@Mock` by calling
+   `MockitoAnnotations.initMocks(testInstance)`
+2. Creates an `IdeaProjectTestFixture`, which sets up static application state for the IntelliJ SDK
+   and creates a `Project`
+
+**Additional Annotations:**
+
+`@MockService`:
+- Annotates fields of any type, but should also have the `@Mock` annotation
+- Before the test runs, it swaps the actual implementation of the service that is registered in the
+  application's `PicoContainer`
+- After the test runs, it swaps the real implementation back
+- Sample:
+    ```
+    @Mock @MockService private CloudSdkService mockCloudSdkService;
+
+    @Test
+    public void myTest() {
+      when(mockCloudSdkService.validateCloudSdk()).thenReturn(ImmutableSet.of());
+      // Do something that depends on this mocked call...
+    }
+    ```
+
+`@TestFixture`:
+- Annotates fields of type `com.intellij.testFramework.fixtures.IdeaProjectTestFixture`
+- Injects the created `IdeaProjectTestFixture` into the annotated field, giving access to the
+  underlying `Project`
+- Sample:
+    ```
+    @TestFixture private IdeaProjectTestFixture testFixture;
+
+    @Test
+    public void myTest() {
+      Project project = testFixture.getProject();
+      // Do something with the project...
+    }
+    ```
+
+`@TestModule`:
```

Comment:
What is the connection between this Module's project (module#getProject) and the project supplied by the test fixture? Are they the same?